docker/nginx/conf.d/client.conf

```diff
@@ -35,6 +39,17 @@ server {
 	client_max_body_size 128k;
 
 	location / {
+		# The only safe thing to do inside an if in a location block is return
+		# or rewrite, since we need to proxy_pass we have to work our way around
+		# using a custom error code.
+		#
+		# See https://www.nginx.com/resources/wiki/start/topics/depth/ifisevil/
+		error_page 418 = @subdomain;
+		recursive_error_pages on;
+		if ($subdomain  != "") {
+			return 418;
+		}
+
```

Comment:
why do we need that, what is this for?

Reply:
I've linked the explanation in the comment. This is the suggested implementation if you want to use the 'if' directive in a `location` block. I think it's a pretty neat way to handle the proxy actually. The fact that it's the proposed solution in the nginx docs makes it even more ok to be using imo.

Reply:
what I mean is why do we need that whole if at all ? Am I not getting the logic here? the `location /` block is for the main webpage and it has to be accessible without the subdomain and this `if` block basically will return `418` in that case?

Reply:
I think we have to add it to this `location` block because the urls won't contain a skylink no more, but rather a base32 encoded skylink as subdomain. Could be that I am mistaken but I tested this config already and it worked perfectly.

To clarify:
this `https://200dd950uidil1uqqpdem6qaf6s3l46pu1h74ff1v5s4fh8ag3dece0.siasky.dev/` needs to hit the `if`
this `https://siasky.dev/` does not,
and I don't think it will.

Reply:
To clarify further, defining `error_page` causes an **internal** redirect to happen. So it's not like the user is seeing that error code, it'll just be redirected internally to a named location. http://nginx.org/en/docs/http/ngx_http_core_module.html#error_page